extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/context/ArcContextProvider.java

```diff
@@ -49,17 +41,31 @@ public ThreadContextSnapshot currentContext(Map<String, String> map) {
                 // context active, store current state, feed it new one and restore state afterwards
                 InjectableContext.ContextState stateToRestore = requestContext.getState();
                 requestContext.deactivate();
-                requestContext.activate(state);
+                if (state != null) {
```

Comment:
OK, that's funny: this is _precisely_ what I said would be easier for API users: if inactive contexts return a `null` state and if passing `null` to `activate` would deactivate the context ;) See, I'm not _that_ crazy after all.

Reply:
I don't think passing `null` into `activate()` to deactivate is intuitive. Especially not when we need to differentiate between deactivate and destroy.

Reply:
Nevermind.